app/src/main/java/com/github/se/cyrcle/model/report/ReportedObjectRepositoryFirestore.kt

```diff
@@ -3,9 +3,51 @@ package com.github.se.cyrcle.model.report
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.gson.Gson
 import com.google.gson.GsonBuilder
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import com.google.gson.JsonSerializationContext
+import com.google.gson.JsonSerializer
 import com.google.gson.reflect.TypeToken
+import java.lang.reflect.Type
 import javax.inject.Inject
 
+/**
+ * A Firestore-backed implementation of the ReportedObjectRepository. Manages CRUD operations for
+ * reported objects in a Firestore database.
+ *
+ * @property db The Firestore instance used for database operations.
+ */
+class ReportReasonAdapter : JsonDeserializer<ReportReason>, JsonSerializer<ReportReason> {
+  override fun serialize(
+      src: ReportReason,
+      typeOfSrc: Type,
+      context: JsonSerializationContext
+  ): JsonElement {
+    val jsonObject = JsonObject()
+    jsonObject.addProperty("type", src::class.java.simpleName)
+    jsonObject.add("data", context.serialize(src))
```

Comment:
I would appreciate if you added a comment on this line. I don't really understand how this works

Reply:
Done! (documentation was added for the serializer)